packages/react-router-dom/modules/index.js

```diff
@@ -7,6 +7,7 @@ export Prompt from "./Prompt";
 export Redirect from "./Redirect";
 export Route from "./Route";
 export Router from "./Router";
+export RouterContext from "./RouterContext";
```

Comment:
Let's not export `RouterContext` as part of our public API. It's an implementation detail that users shouldn't care about. If they want routing data, they can always render a `<Route>` or use `withRouter`.

Reply:
@alexeyraspopov `<Route>` has a render prop API, both with the `render` prop and a children-as-a-function "prop".

Reply:
Oh, I've just found out that `path` is not required. Worth trying, thank you.

Reply:
@mjackson what about a case where an application is using a Router nested inside another Router and the nested Router contains a NavLink that would like to use the parent Router's Context... if the Context is exported it can be explicitly used in this situation. Thoughts?

Reply:
@crobinson42 Is such a thing possible today with the old context system? If not, isn't that out of scope for this PR?

Reply:
@josephcsible I have not been able to find a solution using the old context system. I suppose it would be considered in-scope since this PR would provide the opportunity to use the library in a new way by the benefits of the new context api. But, I do see your point in the requirement it would add to then add a prop to the wrapping `<Router context={SpecificRouterContext}>` component.